File: Go/MergeSort.go
//GO实现归并排序

func mergeSort(arr []int)[]int{
	length := len(arr)
	if length < 2{
		return arr
	}
	middle := length / 2
	left,right := arr[:middle],arr[middle:]
	return merge(mergeSort(left),mergeSort(right))
}

func merge(left,right []int)[]int{
	var result []int
	for len(left)!=0&&len(right)!=0{
		if(left[0] <= right[0]){
			result = append(result,left[0])
			left = left[1:]
		}else{
			result = append(result,right[0])
			right = right[1:]
		}
	}

	for len(left) != 0{
		result = append(result,left[0])
		left = left[1:]
	}
	for len(right) != 0{
		result = append(result,right[0])
		right = right[1:]
	}
	return result
}

File: Python/MergeSort.py
# -*- coding = utf-8 -*-
# Python实现归并排序

def mergeSort(arr):
    import math
    if(len(arr)<2):
        return arr
    middle = math.floor(len(arr)/2)
    left,right = arr[:middle],arr[middle:]
    return merge(mergeSort(left),mergeSort(right))



def merge(left,right):
    result = []
    while left and right:
        if left[0] <= right[0]:
            result.append(left.pop(0))
        else:
            result.append(right.pop(0))
    while left:
        result.append(left.pop(0))
    while right:
        result.append(right.pop(0))
    return result